tests/models/test_case_small_network_optional_assets_annualized/src/run_annualized.py

```diff
@@ -0,0 +1,231 @@
+from rtctools_heat_network.heat_mixin import HeatMixin
+
+
+class TargetDemandGoal(Goal):
+    priority = 1
+    order = 2
+
+    def __init__(self, state, target):
+        self.state = state
+
+        self.target_min = target
+        self.target_max = target
+        self.function_range = (-1.0, 2.0 * max(target.values))
+        self.function_nominal = np.median(target.values)
+
+    def function(self, optimization_problem, ensemble_member):
+        return optimization_problem.state(self.state)
+
+
+class MinimizeDiscAnnualizedCostGoal(Goal):
+
+    """This class represents an optimization goal that minimizes the
+      discounted annualized cost of a heat network model.
+
+    It takes the annualized cost calculated by the model and minimizes
+    it by changing optimization variables such as asset sizes.
+    """
+
+    order = 1
+    priority = 2
+
+    def __init__(self):
+        self.target_max = 0.0
+        self.function_range = (0.0, 1.0e8)
+        self.function_nominal = 1.0e7
+
+    def function(self, optimization_problem: HeatMixin, ensemble_member):
+        obj = 0.0
+
+        assets_and_cost_keys = [
+            (["source", "ates"], ["_asset_variable_operational_cost_map"]),
+            (["source", "ates", "buffer"], ["_asset_fixed_operational_cost_map"]),
+            (
+                [
+                    "source",
+                    "ates",
+                    "buffer",
+                    "demand",
+                    "heat_exchanger",
+                    "heat_pump",
+                    "pipe",
+                ],
+                ["_annualized_capex_var_map"],
+            ),
+        ]
+
+        for asset_categories, cost_map_keys in assets_and_cost_keys:
+            obj += self.sum_cost(optimization_problem, asset_categories, cost_map_keys)
+
+        return obj
+
+    def sum_cost(
+        self,
+        optimization_problem,
+        asset_categories: List[str],
+        cost_map_keys: List[str],
+    ):
+        """
+        Sums up the cost for the given optimization problem, asset categories,
+        and cost map keys.
+        """
+        obj = 0.0
+        for asset_category in asset_categories:
+            for asset in optimization_problem.heat_network_components.get(asset_category, []):
+                for cost_map_key in cost_map_keys:
+                    cost_map = getattr(optimization_problem, cost_map_key)
+                    cost = cost_map.get(asset, 0)
+                    obj += optimization_problem.extra_variable(cost)
+        return obj
+
+
+class _GoalsAndOptions:
+    def path_goals(self) -> List[Goal]:
+        goals = super().path_goals().copy()
+
+        for demand in self.heat_network_components["demand"]:
+            target = self.get_timeseries(f"{demand}.target_heat_demand")
+            state = f"{demand}.Heat_demand"
+
+            goals.append(TargetDemandGoal(state, target))
+
+        return goals
+
+
+class HeatProblem(
```

Comment:
If I see it correctly this HeatProblem is a direct copy of the HeatProblem in run_ates.py. If this is the case, I recommend that you remove it in one of the two files and import it from the file where it is defined instead.

Reply:
Good point!